policy/modules/kernel/kernel.te

```diff
@@ -95,6 +95,10 @@ type proc_kcore_t, proc_type;
 neverallow ~{ can_dump_kernel kern_unconfined } proc_kcore_t:file ~getattr;
 genfscon proc /kcore gen_context(system_u:object_r:proc_kcore_t,mls_systemhigh)
 
+optional_policy(`
+	init_mountpoint(proc_kcore_t)
```

Comment:
True, this looks like a clear violation of
https://github.com/SELinuxProject/refpolicy/blob/master/policy/modules/kernel/kernel.te#L95

But this should have been catched by CI, and running the `validate` target locally doesn't generate this error.

Reply:
So, we set 'expand-check=1' in /etc/selinux/semange.conf. I'm pretty sure that is why the error is showing up for me. I guess that CI does't have a similar setting.  Or maybe it can be enabled to catch this kind of thing in the future.

I guess the real question is, what is the correct solution?
A new attribute that can be added to the neverallow list?
It doesn't seem like we want to add init_t to either of the two attributes there.

Or change the name of the 'can_dump_kernel' attribute to be a bit more generic 'kernel_kcore_access', create an interface to add types to the attribute ('kernel_access_proc_kcore'), init_t can use this new interface.  Finally alter the current 'kernel_read_core_if' interface call this new interface for existing compatibility.

I can submit a patch if desired.

Reply:
The `validate` target uses `semodule_link` and `semodule_expand` to check the policy, it behaves differently (in probably a bad way) than `semodule`.
See also #275 as the result of CI breaking on my RBAC branch, despite *everything* being fine on a live test system.
It would be very nice if `semodule_link` and `semodule_expand` would be made to behave *exactly* like `semodule` or if they were replaced with `semodule` in the `validate` target.

> Or change the name of the 'can_dump_kernel' attribute to be a bit more generic 'kernel_kcore_access', create an interface to add types to the attribute ('kernel_access_proc_kcore'), init_t can use this new interface. Finally alter the current 'kernel_read_core_if' interface call this new interface for existing compatibility.

I would prefer this, since there are other applications that might mount over `proc_kcore_t`, e.g. docker.

Reply:
This isn't a bug in the tools, it is a gap in the CI testing. It's because `init_t` is unconfined:
```
$ make MONOLITHIC=y QUIET=y APPS_OFF=unconfined SYSTEMD=y bare
$ make MONOLITHIC=y QUIET=y APPS_OFF=unconfined SYSTEMD=y conf
Generating interface templates into doc/tmp/iftemplates
Creating policy.xml
Updating policy/booleans.conf and policy/modules.conf
$ make MONOLITHIC=y QUIET=y APPS_OFF=unconfined SYSTEMD=y policy
Creating refpolicy policy.conf
Compiling refpolicy policy.32
libsepol.report_failure: neverallow on line 95 of policy/modules/kernel/kernel.te (or line 20736 of policy.conf) violated by allow init_t proc_kcore_t:file { mounton };
libsepol.check_assertions: 1 neverallow failures occurred
Error while expanding policy
make: *** [Rules.monolithic:68: policy.32] Error 1
$ make MONOLITHIC=n QUIET=y APPS_OFF=unconfined SYSTEMD=y validate
Compiling refpolicy base module
libsepol.report_failure: neverallow on line 95 of policy/modules/kernel/kernel.te (or line 20736 of policy.conf) violated by allow init_t proc_kcore_t:file { mounton };
libsepol.check_assertions: 1 neverallow failures occurred
/usr/bin/checkmodule:  expand module failed
make: *** [Rules.modular:103: tmp/base.mod] Error 1
```

Reply:
I added additional CI tests in fbdb3755 which will uncover this assertion failure.

Reply:
Thanks Chirs, I have a proposed fix I will push shortly.